fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/mixin/content/registry/FarmerVillagerTaskMixin.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.content.registry;
+
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.ai.brain.task.FarmerVillagerTask;
+import net.minecraft.entity.passive.VillagerEntity;
+import net.minecraft.inventory.SimpleInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.sound.SoundCategory;
+import net.minecraft.util.math.BlockPos;
+
+import net.fabricmc.fabric.api.registry.VillagerPlantableRegistry;
+
+@Mixin(FarmerVillagerTask.class)
+public class FarmerVillagerTaskMixin {
+	@Nullable
+	@Shadow private BlockPos currentTarget;
+
+	@Inject(method = "keepRunning", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/passive/VillagerEntity;getInventory()Lnet/minecraft/inventory/SimpleInventory;"))
+	private void fabric_useRegistry(ServerWorld serverWorld, VillagerEntity villagerEntity, long l, CallbackInfo ci) {
```

Comment:
I'm not too fond of this mixin, it is essentially deleting vanilla behavior and replacing it. This might break mods that mixin somewhere in these checks, and most importantly it might break vanilla behavior when updating to a new version. I would prefer a modify variable hook after the if chain, if possible.

Reply:
After taking another look, I have a new ModifyVariable mixin on the read of the if statement, which should be much safter.
It also leaves the vanilla crops untouched, so they are no longer included in the registry.
Let me know what you think of this solution.